pages/docs/modules/exec.mdx

```diff
@@ -0,0 +1,355 @@
+# exec
+
+## Functions
+
+### command
+
+```go filename="Function signature"
+command(file string, args ...string) exec.command
+```
+
+Creates a new `exec.command` targeting a specific `file` to execute,
+followed by arguments to execute the command with.
+
+Nothing is executed by this function alone, but instead only a [`exec.command`](#type-command)
+object is returned. To execute `file`, you must call the [`run`](#commandrun)
+method on the [`exec.command`](#type-command) object.
+
+The lookup if `file` executable exists is also not performed until the [`run`](#commandrun)
+method is called.
+
+```go copy filename="Example"
+>>> exec.command("bash")
+exec.command("bash")
+>>> exec.command("bash", "-c", "echo hello world")
+exec.command("bash", "-c", "echo hello world")
+>>> exec.command("file-that-does-not-exist")
+exec.command("file-that-does-not-exist")
+>>> cmd := exec.command("date", "-Iseconds")
+>>> cmd.run()
+>>> cmd.stdout
+"2024-01-09T21:33:48+01:00\n"
+```
+
+### look_path
+
+```go filename="Function signature"
+look_path(file string) string
+```
+
+Searches for an executable named `file` in the directories named by the `PATH`
+environment variable.
+If `file` contains a slash, it is tried directly and the `PATH` is not consulted.
+Otherwise, on success, the result is an absolute path.
+
+```go copy filename="Example"
+>>> exec.look_path("bash")
+"/usr/bin/bash"
+```
+
+### exec
+
+```go filename="Function signature"
+exec(file string, args list = [], params map = {}) exec.result
+```
```

Comment:
Maybe not right place to discuss module design, but what about making the `exec` function also having variadic parameters, so it gets identical parameters as `exec.command`?

Would make sense to me to have `exec` only execute, with very limited extra options. Then if you need more control, such as changing environment variables, then you have to switch over to use `exec.command`.

Reply:
I don't love what's there right now on exec, so I'm open to adjustments.

Reply:
On the other hand, as Risor doesn't have "exploding operator" (lack for a better term), then using variadic functions actually becomes quite tricky to use dynamically.

For example, what if you wanted to execute a command with optional flags. Such as:

```go
args := ["install", "my-repo/my-chart"]

if should_run_dry_mode() {
  args.append("--dry-run")
}

exec("helm", args)
```

Doing the above with variadic parameters is not possible (without using the `call` builtin).

Could've been possible if Risor had syntax like this:

```go
exec("helm", ...args)
```

But it doesn't (yet?). Question is which one to design for?